tests/vmi_multus_test.go

```diff
@@ -986,16 +1025,76 @@ var _ = Describe("[Serial]Macvtap", func() {
 
 		BeforeEach(func() {
 			macAddress := "02:03:04:05:06:07"
-			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+			clientVMI, err = createCirrosVMIRandomNode(macvtapNetworkName, macAddress)
+			Expect(err).NotTo(HaveOccurred(), "must succeed creating a VMI on a random node")
 		})
 
 		It("should be successful when the VMI MAC address is defined in its spec", func() {
 			By("starting the migration")
-			migration := tests.NewRandomMigration(macvtapVMI.GetName(), macvtapVMI.GetNamespace())
+			migration := tests.NewRandomMigration(clientVMI.GetName(), clientVMI.GetNamespace())
 			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
 
 			// check VMI, confirm migration state
-			tests.ConfirmVMIPostMigration(virtClient, macvtapVMI, migrationUID)
+			tests.ConfirmVMIPostMigration(virtClient, clientVMI, migrationUID)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverIP string
+
+			getVMMacvtapIfaceIP := func(vmi *v1.VirtualMachineInstance, macAddress string) (string, error) {
+				var vmiIP string
+				err := wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+					vmi, err := virtClient.VirtualMachineInstance(vmi.GetNamespace()).Get(vmi.GetName(), &v13.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+
+					for _, iface := range vmi.Status.Interfaces {
+						if iface.MAC == macAddress {
+							vmiIP = iface.IP
+							return true, nil
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				// code below should be removed if / when KubeVirt honors its
+				// API and returns an IP on the Interface.IP attribute
+				// Track the following PR to know when to update:
+				//   - https://github.com/kubevirt/kubevirt/pull/4217
+				ip, _, err := net.ParseCIDR(vmiIP)
+				if err != nil {
+					return "", fmt.Errorf("could not compute the IP address from the CIDR. Reason: %v", err)
+				}
+				return ip.String(), nil
+			}
+
+			BeforeEach(func() {
+				macAddress := "02:03:04:05:06:aa"
+
+				serverVMI, err = createFedoraVMIRandomNode(macvtapNetworkName, macAddress)
+				Expect(err).NotTo(HaveOccurred(), "must have succeeded creating a fedora VMI on a random node")
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+
+				serverIP, err = getVMMacvtapIfaceIP(serverVMI, macAddress)
+				Expect(err).NotTo(HaveOccurred(), "should have managed to figure out the IP of the server VMI")
+			})
+
+			BeforeEach(func() {
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *before* migrating the VMI")
+			})
+
+			FIt("should keep connectivity after a migration", func() {
+				migration := tests.NewRandomMigration(serverVMI.GetName(), serverVMI.GetNamespace())
```

Comment:
nit: I'd prefer to be consistent with the other usages of `Name` and `Namespace`. Please use the attribute and not the functions.

Reply:
Done.